dart/lib/src/scope.dart

```diff
@@ -390,29 +390,29 @@ class Scope {
 
     await clone.setUser(user);
 
-    for (final tag in _tags.keys) {
+    for (final tag in List.from(_tags.keys)) {
       await clone.setTag(tag, _tags[tag]!);
     }
 
-    for (final extraKey in _extra.keys) {
+    for (final extraKey in List.from(_extra.keys)) {
       await clone.setExtra(extraKey, _extra[extraKey]);
     }
 
-    for (final breadcrumb in _breadcrumbs) {
+    for (final breadcrumb in List.from(_breadcrumbs)) {
       await clone.addBreadcrumb(breadcrumb);
     }
 
-    for (final eventProcessor in _eventProcessors) {
+    for (final eventProcessor in List.from(_eventProcessors)) {
       clone.addEventProcessor(eventProcessor);
     }
 
-    contexts.forEach((key, value) {
+    Map.from(contexts).forEach((key, value) async {
       if (value != null) {
-        clone.setContexts(key, value);
+        await clone.setContexts(key, value);
       }
     });
```

Comment:
@denrase that has to be a sync call, so the solution should rather be:

```
for (final entry in contexts.entries) {
      if (entry.value != null) {
        await clone.setContexts(entry.key, entry.value);
      }
    }
```

Otherwise each item of the loop is async but we don't await to its future.